s4e-backend/src/test/java/pl/cyfronet/s4e/controller/InvitationControllerTest.java

```diff
@@ -128,6 +158,44 @@ public void shouldCreate() throws Exception {
         assertEquals(dbInvitation.get().getStatus(), InvitationStatus.WAITING);
     }
 
+    @Test
+    public void resendShouldBeSecured() throws Exception {
+        val request = InvitationHelper.invitationRequestBuilder().build();
+        val invitation = invitationRepository
+                .save(InvitationHelper.invitationBuilder(institution).build());
+        val URL = API_PREFIX_V1 + "/invitation/{token}";
+        mockMvc.perform(put(URL, invitation.getToken())
+                .contentType(MediaType.APPLICATION_JSON)
+                .with(jwtBearerToken(member, objectMapper))
+                .content(objectMapper.writeValueAsBytes(request))
+        ).andExpect(status().isForbidden());
+    }
+
+    @Test
+    public void shouldResend() throws Exception {
```

Comment:
Apart from this, I would test if an attempt to resend an invitation from another institution fails.
Case would look as follows: two institutions A and B, in B there is an invitation b. institutionAdmin is an admin to A but not B, and he tries to do `PUT /institutions/A/invitation/b` → I think it should effect in 400, as access to A is assured, but there is no match between A and b.